onnx/defs/nn/defs.cc

```diff
@@ -340,12 +340,22 @@ void maxUnpoolShapeInference(InferenceContext& ctx) {
   if (hasNInputShapes(ctx, 3)) {
```

Comment:
This check is not the same as checking if the optional 3rd input is specified. If the optional 3rd input is specified, we will need to return, I believe.

Reply:
Fixed.